sdk/servicebus/Azure.Messaging.ServiceBus/tests/ReceiverLiveTests.cs

```diff
@@ -48,5 +47,52 @@ await foreach (ServiceBusMessage peekedMessage in peekedMessages)
                 Assert.AreEqual(messageCt, ct);
             }
         }
+
+        [Test]
+        public async Task ReceiveMessagesInPeekLockMode()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false))
+            {
+                await using var sender = new QueueSenderClient(TestEnvironment.ServiceBusConnectionString, scope.QueueName);
+                await sender.SendRangeAsync(GetMessages(10));
+
+                var receiver = new QueueReceiverClient(TestEnvironment.ServiceBusConnectionString, scope.QueueName);
+                var messageCount = 0;
+
+                await foreach (var item in receiver.ReceiveBatchAsync(10))
+                {
+                    messageCount++;
+                    Assert.AreEqual(item.SystemProperties.DeliveryCount, 1);
+                }
+                Assert.AreEqual(messageCount, 10);
+            }
+        }
+
+        [Test]
+        public async Task ReceiveMessagesInReceiveAndDeleteMode()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false))
+            {
+                await using var sender = new QueueSenderClient(TestEnvironment.ServiceBusConnectionString, scope.QueueName);
+                await sender.SendRangeAsync(GetMessages(10));
+
+                var clientOptions = new QueueReceiverClientOptions()
+                {
+                    ReceiveMode = ReceiveMode.ReceiveAndDelete,
+                };
+                var receiver = new QueueReceiverClient(TestEnvironment.ServiceBusConnectionString, scope.QueueName, clientOptions);
+                var messageCount = 0;
+
+                await foreach (var item in receiver.ReceiveBatchAsync(10))
```

Comment:
Should we try to receive more than we sent to verify that only our sent messages are received?
Or track the message Ids?

Reply:
Updated the test cases to track message Ids.